specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/stable/2016-06-01/vaults.json

```diff
@@ -589,6 +644,47 @@
       },
       "x-ms-azure-resource": true
     },
+    "ResourceNameAvailabilityInput": {
+      "description": "Resource information, as sent to the regional resource provider from Global RP.",
+      "type": "object",
+      "properties": {
+        "type": {
+          "type": "string"
+        },
+        "name": {
+          "type": "string"
+        }
+      }
+    },
+    "ResourceNameAvailabilityResponse": {

```

Comment:
I don't think this needs to be its own model, can't these fields be embedded directly in the parent?

Reply:
The fields present in the response are completely different to that in the parent.